hbase-zookeeper/src/main/java/org/apache/hadoop/hbase/zookeeper/ZKWatcher.java

```diff
@@ -460,6 +460,9 @@ public List<String> getMetaReplicaNodes() throws KeeperException {
   public List<String> getMetaReplicaNodesAndWatchChildren() throws KeeperException {
     List<String> childrenOfBaseNode =
       ZKUtil.listChildrenAndWatchForNewChildren(this, znodePaths.baseZNode);
+    if (childrenOfBaseNode == null) {
```

Comment:
Better to add a comment to say that why we need to throw an exception here instead of returning an empty list, so later developers will not remove it since there is no tests for this...

Reply:
Agreed, updated with comment